traveler-ios-ui/TravelerKitUI/Models/ParkingFilterContext.swift

```diff
@@ -15,21 +15,37 @@ public protocol ParkingFilterContextObserving: class {
 public class ParkingFilterContext {
     private(set) var filter: Int?
 
-    private var delegate: ParkingFilterContextObserving?
+    private weak var delegate: ParkingFilterContextObserving?
 
     var selectedFilter: Int? {
         didSet {
-            observers.forEach { $0.parkingFilterContextDidChangeSelectedFilter(self)}
+            notifyObserver()
         }
     }
 
-    private var observers = [ParkingFilterContextObserving]()
+    private struct Observation {
```

Comment:
The observer logic should be abstracted.  Lots of repetition where only the Observing type differs.  Make one generic Observing type and utilize in specific instances.